pandas/_libs/window.pyx

```diff
@@ -1391,7 +1392,19 @@ def roll_quantile(ndarray[float64_t, cast=True] input, int64_t win,
 
         if nobs >= minp:
             idx = int(quantile * <double>(nobs - 1))
-            output[i] = skiplist.get(idx)
+
+            # Exactly last point
+            if idx == nobs - 1:
+                output[i] = skiplist.get(idx)
+
+            # Interpolated quantile
+            else:
+                qlow = (<double> idx) / (<double>(nobs - 1))
```

Comment:
The branch that handled quantile == 1.0 also handled nobs == 1, that now is handled separately in the innermost loop. There may be more optimal ways to handle this case in a more external loop, but IMHO this change is not that bad, since this condition only happens when there's a single loop.

I have not added more tests, since TestMoments::test_rolling_quantile tests the mentioned cases too.